blackdoc/formats/__init__.py

```diff
@@ -0,0 +1,31 @@
+import textwrap
+
+from . import doctest, none
+from .register import detection_funcs  # noqa
+from .register import extraction_funcs, reformatting_funcs, register_format
+
+
+def extract_code(line_unit, category):
+    dedented = textwrap.dedent(line_unit)
+    indentation_depth = line_unit.find(dedented[:5])
+
+    func = extraction_funcs.get(category, None)
+    if func is None:
+        raise RuntimeError(f"unknown code format: {category}")
```

Comment:
it seems a bit inconsistent to call them both a format and a category. Maybe use `code_format` instead?